contrib/namespace-type-lib/test.ts

```diff
@@ -0,0 +1,36 @@
+import { Namespace, SubjectSet, Context } from "@ory/keto-namespace-types"
+
+// This test is not really a valid config, but rather a check of the types.
+class User implements Namespace {
+  related: {
+    friends: User[]
+  }
+}
+
+class Group implements Namespace {
+  related: {
+    members: (User | Group)[]
+  }
+
+  permits = {
+    isMember: (ctx: Context): boolean =>
+      this.related.members.traverse((m) =>
+        m instanceof User ? m == ctx.subject : m.permits.isMember(ctx),
```

Comment:
Checking by `instanceof` is not part of the OPL. I agree that there is some usefulness, but shouldn't we implement it before we put it into tests?

Reply:
The test is only to ensure that the typing works. With this `instanceof` conditional I could verify that the members are actually either of type `User` or `Group`. It really only is about the type library correctness.